packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -0,0 +1,401 @@
+[prosemirror-markdown](https://github.com/ProseMirror/prosemirror-markdown) serializer. Here is the reference for the supported formatting schema in the
+markdown based on [CommonMark](http://commonmark.org/) flavor:
+
+-   Bold - `**Bold**`
+-   Italics - `*Italics*`
+-   Numbered list - `1. Numbered list`
+-   Bulleted list - `* Bulleted list`
+
+The `nimble-rich-text-viewer` will be responsible for converting the input markdown string to HTML Fragments with the help of
+`prosemirror-markdown` parser, which is then converted to HTML string and rendered into the component to view all rich text content.
+
+### Prototype
+
+[`Tiptap prototype in Stackblitz`](https://stackblitz.com/edit/typescript-g3yfmo?file=index.ts)
+
+The prototype includes the below functionalities,
+
+1. Basic formatting support.
+2. Image support. - The prototype includes [`Tiptap Extension`](https://tiptap.dev/api/nodes/image) to support images.
+3. `@mention` support. - The prototype includes a default user list that gets triggered using `@` character.
+4. Markdown support. - The prototype includes support to render/retrieve content in markdown format.
+5. `Shadow root` support - The prototype uses [Microsoft Fast](https://github.com/microsoft/fast) to create the rich text editor as a
+   custom component that renders in the shadow root.
+6. `Top layer` support - Hyperlink using `nimble-dialog` which renders the popup in the top layer.
+
+**_Note_**: This prototype is not made in a responsive fashion for smaller screens and is intended only for testing the specific functionalities listed
+above.
+
+### States
+
+_Button State_
+
+The buttons in the toolbar will indicate the formatting status of the selected text in the editor. We use the
+[nimble-toggle-button](https://nimble.ni.dev/storybook/?path=/docs/toggle-button--docs) to determine whether it's in an active or inactive state.
+Below is the sample screenshot which shows the active state of the `nimble-toggle-button` when the cursor is focused on the `Bold` and `Numbered` List`
+text in the editor.
+
+![Bold and Numbered Button State](./spec-images/button-state.png)
+
+### Accessibility
+
+[Tiptap Accessibility](https://tiptap.dev/guide/accessibility)
+
+[Toolbar Accessibility](https://www.w3.org/WAI/ARIA/apg/patterns/toolbar/)
+
+[Button Accessibility](https://www.w3.org/WAI/ARIA/apg/patterns/button/)
+
+_Focus_
+
+-   Focus state of the editor will be the same as the `nimble-text-area`.
+-   Focus state of the buttons will be the same as the `nimble-button`.
+
+_Keyboard accessibility and shortcuts for text formatting_
+
+All the basic keyboard accessibility for the tiptap rich text editor is mentioned in their
+[keyboard shortcut](https://tiptap.dev/api/keyboard-shortcuts#predefined-keyboard-shortcuts) page. To facilitate easy access, listed down
+the supported text formatting keyboard shortcuts in the tiptap editor.
+
+| Key                     | Behavior                                                 |
```

Comment:
Double checking: does TipTap automatically support Command instead of Ctrl for the shortcuts on macOS?

Reply:
Yes, it does support `Cmd` in macOS, and here is the official page of Tiptap for [keyboard shortcuts](https://tiptap.dev/api/keyboard-shortcuts#predefined-keyboard-shortcuts). I have added this point as well in the spec.